mount/mount_windows.go

```diff
@@ -85,25 +131,52 @@ func (m *Mount) GetParentPaths() ([]string, error) {
 
 // Unmount the mount at the provided path
 func Unmount(mount string, flags int) error {
-	var (
-		home, layerID = filepath.Split(mount)
-		di            = hcsshim.DriverInfo{
-			HomeDir: home,
+	mount = filepath.Clean(mount)
+	adsFile := mount + ":" + sourceStreamName
+	var layerPath string
+
+	if _, err := os.Lstat(adsFile); err == nil {
+		layerPathb, err := os.ReadFile(mount + ":" + sourceStreamName)
+		if err != nil {
+			return fmt.Errorf("failed to retrieve source for layer %s: %w", mount, err)
 		}
-	)
-
-	if err := hcsshim.UnprepareLayer(di, layerID); err != nil {
-		return fmt.Errorf("failed to unprepare layer %s: %w", mount, err)
+		layerPath = string(layerPathb)
 	}
-	if err := hcsshim.DeactivateLayer(di, layerID); err != nil {
-		return fmt.Errorf("failed to deactivate layer %s: %w", mount, err)
+
+	if err := bindfilter.RemoveFileBinding(mount); err != nil {
+		if errno, ok := errors.Unwrap(err).(syscall.Errno); ok && errno == windows.ERROR_INVALID_PARAMETER || errno == windows.ERROR_NOT_FOUND {
```

Comment:
Wait, can we do `errors.Is(err, windows.ERROR_INVALID_PARAMETER) || errors.Is(err, windows.ERROR_NOT_FOUND)`?

Reply:
I think so, yeah. We couldn't for the hcsshim-supplied errors (see https://github.com/microsoft/hcsshim/issues/960) but go-winio's bindfilter functions are just using `fmt.Errorf` with `%w` for the errors coming from mkwinsyscall, so `errors.Is` should work naturally. AFAIR, the `windows.ERROR_*` constants do work correctly with `errors.Is`.